------------------------------------------------------------
revno: 14144
revision-id: squid3@treenet.co.nz-20170226084624-5tkl3bdrqz8nlp9g
parent: squid3@treenet.co.nz-20170225055014-j7v5xax13u4jddr9
author: Alex Rousskov <rousskov@measurement-factory.com>
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.5
timestamp: Sun 2017-02-26 21:46:24 +1300
message:
  Fix crash when configuring with invalid delay_parameters restore value.
  
  ... like none/none. Introduced in rev which fixed another, much
  bigger delay_parameters parsing bug.
  
  TODO: Reject all invalid input, including restore/max of "-/100".
  
  TODO: Fix misleading/wrong associated error messages. For example:
    ERROR: invalid delay rate 'none/none'. Expecting restore/max or 'none'
    ERROR: restore rate in '1/none' is not a number.
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20170226084624-5tkl3bdrqz8nlp9g
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# testament_sha1: 42f47b8ee1da049d57e6af76ce755e459d2fc9fd
# timestamp: 2017-02-26 08:51:02 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.5
# base_revision_id: squid3@treenet.co.nz-20170225055014-\
#   j7v5xax13u4jddr9
# 
# Begin patch
=== modified file 'src/DelaySpec.cc'
--- src/DelaySpec.cc	2017-01-01 00:16:45 +0000
+++ src/DelaySpec.cc	2017-02-26 08:46:24 +0000
@@ -55,7 +55,7 @@
 
     // parse the first digits into restore_bps
     const char *p = NULL;
-    if (!StringToInt(token, restore_bps, &p, 10) && *p != '/') {
+    if (!StringToInt(token, restore_bps, &p, 10) || *p != '/') {
         debugs(77, DBG_CRITICAL, "ERROR: invalid delay rate '" << token << "'. Expecting restore/max or 'none'.");
         self_destruct();
     }

